app/src/auth.ts

```diff
@@ -0,0 +1,192 @@
+import { Auth0Plugin, createAuth0 } from "@auth0/auth0-vue";
+import { App as CapApp } from "@capacitor/app";
+import { Browser } from "@capacitor/browser";
+import { Capacitor } from "@capacitor/core";
+import { type App, watch } from "vue";
+import type { Router } from "vue-router";
+import AuthBrowser from "./util/authBrowser";
+
+const appId = import.meta.env.VITE_CAP_APPID;
+const authDomain = import.meta.env.VITE_AUTH0_DOMAIN;
+
+export type AuthPlugin = Auth0Plugin & {
+    logout: (retrying?: boolean) => Promise<void>;
+};
+
+export default async function setupAuth(app: App<Element>, router: Router) {
+    const app_scheme = `${appId}://${authDomain}/capacitor/${appId}`;
+    const web_origin = window.location.origin;
+    const platform = Capacitor.getPlatform();
+
+    const oauth = createAuth0(
+        {
+            domain: authDomain,
+            clientId: import.meta.env.VITE_AUTH0_CLIENT_ID,
+            useRefreshTokens: true,
+            useRefreshTokensFallback: platform === "web",
+            cacheLocation: "localstorage",
+            authorizationParams: {
+                audience: import.meta.env.VITE_AUTH0_AUDIENCE,
+                scope: "openid profile email offline_access",
+                redirect_uri: platform === "web" ? web_origin : `${app_scheme}/callback`,
+            },
+        },
+        {
+            skipRedirectCallback: true,
+        },
+    );
+
+    // Handle redirects (Save token to local storage)
+    async function redirectCallback(_url: string) {
+        const url = new URL(_url);
+        console.log(url);
+        if (
+            url.searchParams.has("state") &&
+            (url.searchParams.has("code") || url.searchParams.has("error"))
+        ) {
+            if (url.searchParams.has("error")) {
+                const error = url.searchParams.get("error");
+                console.error(error);
+                alert(error);
+            } else {
```

Comment:
Instead of using an "else" clause, you can use an "early return" coding style which improves readability. We did it throughout the project. It is not always practical, but can be used in most cases and prevents a very deep nested if structure which becomes hard to read.